### PR TITLE
resource/key_pair: drop custom ValidateFunc for name_prefix

### PR DESCRIPTION
**Pull request by loivis**
It's a serie of PRs to drop custom `ValidateFunc`. The goal is to use existing functions in `validation` package as much as possible. To minimize the scope and make it easy for review, one PR will be created per resource or data source.

Acceptance tests might fail as I didn't run any of them at all. I'll fix it if you find any error during review.

This PR is for:

- [x] resource/key_pair

### CODE REVIEW

File: aws/resource_aws_key_pair.go [hunk 1]
```diff
@@ -32,19 +32,13 @@ func resourceAwsKeyPair() *schema.Resource {
...
-					if len(value) > 100 {
-						errors = append(errors, fmt.Errorf(
-							"%q cannot be longer than 100 characters, name is limited to 255", k))
-					}
-					return
-				},
+				Type:         schema.TypeString,
+				Optional:     true,
+				ForceNew:     true,
+				ValidateFunc: validateMaxLength(100),
```
Comment (paultyng) [inline aws/resource_aws_key_pair.go]: can you add a comment here about the 255 limit

Comment (loivis) [reply aws/resource_aws_key_pair.go]: I would rather change it to `validateMaxLength(255 - 26)` but was unsure if `100` was there for some reason. What do you think?

**Comment by paultyng** [comment]
FYI I put up a PR to [helper/resource](https://github.com/hashicorp/terraform/pull/17551) that would remove the need to hard code these lengths and the comments, but probably won't land for a little.

**Comment by paultyng** [comment]
Also, thanks a lot for cleaning all this up!

**Comment by loivis** [comment]
That's neat. It's always better with self-explanatory codes than comments. I'll wait to sync the change when it's merged.

**Comment by loivis** [comment]
sync with merged pr https://github.com/hashicorp/terraform/pull/17551 and https://github.com/terraform-providers/terraform-provider-aws/pull/3798

**Comment by paultyng** [comment]
LGTM, I'll let @bflad merge.

File: aws/resource_aws_key_pair.go [hunk 2]
```diff
@@ -26,27 +26,21 @@ func resourceAwsKeyPair() *schema.Resource {
...
-						errors = append(errors, fmt.Errorf(
-							"%q cannot be longer than 100 characters, name is limited to 255", k))
-					}
-					return
-				},
+			"key_name_prefix": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				ForceNew:     true,
+				ValidateFunc: validateMaxLength(255 - resource.UniqueIDSuffixLength),
```
Comment (bflad) [inline aws/resource_aws_key_pair.go]: 😍

**Pull request merged**